src/NUnitFramework/framework/Constraints/EqualityAdapter.cs

```diff
@@ -24,11 +24,18 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
-using System.Reflection;
 using NUnit.Framework.Compatibility;
+using System.Reflection;
 
 namespace NUnit.Framework.Constraints
 {
+
+#if NET_2_0
+
+    public delegate TResult Func<in T1, in T2, out TResult>(T1 arg1, T2 arg2); 
```

Comment:
I was wondering how Func was working in .NET 2.0. This is a good idea, but I would like to do more of this. This should these be put in in the `NUnit.Compatibility` namespace so people know where it comes from and where to find it. Maybe add a `Net20Compatibility.cs` to `nunit/src/NUnitFramework/framework/Compatibility/` and add this there? You could also pull the `ExtentsionAttribute` out of `nunit/src/NUnitFramework/framework/Compatibility/ReflectionExtensions.cs` since it doesn't really belong there.

I would also be tempted to put `Func` in the same namespace as in the .NET Framework.